### ISSUE TITLE
download failure should return error code

### ISSUE THREAD

**Issue by larrycai** [JFrogDev/jfrog-cli-go#58]
So far the `jfrog rt down` command always return `0` as success

<pre>
$ jfrog --version
jfrog version 1.8.0
$ jfrog rt download none-exist
[Info] Searching items to download...
[Info] Downloaded 0 artifacts.
$ echo $?
0
</pre>

It is not so convient to use. I have to add check whether file exists or not after that.

**Comment by eyalbe4** [comment]
Thanks @larrycai. 
We'll probably need to improve this behavior. We'll let you know after we make the change.

**Comment by larrycai** [comment]
Any deliver plan for this ? it is quite annonying to add extra bash step to check whether it is successful or not ?

**Comment by romangurevitch** [comment]
@larrycai,
Latest JFrog CLI version 1.13.1 was released and includes new summary json.
With the new summary you can check whether the download/upload was executed as expected.
Any errors during download/upload will result in none zero exit code. 
We would appreciate your feedback on this.

**Comment by larrycai** [comment]
@romangurevitch thank for updates

<pre>
$ ./jfrog  rt download  non-exist
[Info] Searching items to download...
{
  "status": "success",
  "totals": {
    "success": 0,
    "failure": 0
  }
}
$ echo $?
0
</pre>

good to see the json result indicates the success download is zero even the result `$?` is ok,

I guess the error code will be non-zero only if the downloading itself is error. It will not show error if I try to download the non-exist file.

If it is the case, please state clearly in the documentation, otherwise it will confuse the user

**Comment by larrycai** [comment]
I still prefer to get non-zero for `$?` when the file can't be downloaded if it doesn't exist.

**Comment by florent-vial** [comment]
I agree with @larrycai . Are there any plans to return non-0 for non-existing path during download?
If not, is this possible to achieve this with the REST API ?

**Comment by eyalbe4** [comment]
@florent-vial, @larrycai,
This is an interesting dilemma. I'd appreciate your insights about this:
On the one hand, the reason why you need a non zero exit code is clear - if no files were uploaded (downloaded, copied, moved...), a non-zero exit code serves your use case.
On the other hand, for other use cases, it may be completely valid to get zero files. For example, a scenario which needs to download specific files if they exist.
In the meantime,  here's a suggested solution that may work for you. Please try it out and let us know what you think. 
Let's create a file named *runJfrog.sh* and put the following lines in it (you'll need to have *jq* installed before running it):
```
#!/bin/bash

if [ "$(${1} | jq '.totals.success')" -eq 0 ]; then 
  exit 1; 
fi 
```
You can then run this script as follows. It will execute: jfrog rt u *.pom repo
```
$ ./runJfrog.sh "jfrog rt u *.pom repo"
```
Running *echo $?* will prompt *1* now, because the jfrog rt u *.pom repo command uploaded zero files and returned the following summary:
```
{
  "status": "failure",
  "totals": {
    "success": 0,
    "failure": 1
  }
}
```

**Comment by florent-vial** [comment]
"On the other hand, for other use cases, it may be completely valid to get zero files. For example, a scenario which needs to download specific files if they exist."

Then this should be a configuration flag in the tool IMO. A default "automation friendly" behaviour should allow to detect errors as soon as possible.

**Comment by eyalbe4** [comment]
@florent-vial and @larrycai,
Version 1.14.0 has been released and includes a new *fail-no-op* option. Adding this option with a *true* value, will cause the command to return with exit code 2, in case it was successful, but no files were effected (zero files were found, uploaded, downloaded, deleted, etc.).
Please give it a try and share with us your feedback.
